PEP8 in segment

dipy/segment/clustering.py:76:80: E501 line too long (80 > 79 characters)
dipy/segment/clustering.py:219:80: E501 line too long (81 > 79 characters)
dipy/segment/clustering.py:267:80: E501 line too long (101 > 79 characters)
dipy/segment/clustering.py:272:80: E501 line too long (118 > 79 characters)
dipy/segment/clustering.py:404:80: E501 line too long (93 > 79 characters)
dipy/segment/clustering.py:425:80: E501 line too long (80 > 79 characters)
dipy/segment/clustering.py:436:80: E501 line too long (88 > 79 characters)
dipy/segment/clustering.py:463:80: E501 line too long (93 > 79 characters)
dipy/segment/mask.py:237:80: E501 line too long (84 > 79 characters)
dipy/segment/mask.py:261:80: E501 line too long (82 > 79 characters)
dipy/segment/mask.py:269:1: E302 expected 2 blank lines, found 1
dipy/segment/mask.py:271:80: E501 line too long (82 > 79 characters)
dipy/segment/mask.py:291:80: E501 line too long (85 > 79 characters)
dipy/segment/quickbundles.py:8:1: W191 indentation contains tabs
dipy/segment/quickbundles.py:8:4: E128 continuation line under-indented for visual indent
dipy/segment/quickbundles.py:12:1: E101 indentation contains mixed spaces and tabs
dipy/segment/quickbundles.py:12:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:13:22: E231 missing whitespace after ','
dipy/segment/quickbundles.py:13:29: E231 missing whitespace after ','
dipy/segment/quickbundles.py:13:41: E231 missing whitespace after ','
dipy/segment/quickbundles.py:14:43: W291 trailing whitespace
dipy/segment/quickbundles.py:15:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:20:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:21:26: W291 trailing whitespace
dipy/segment/quickbundles.py:23:19: W291 trailing whitespace
dipy/segment/quickbundles.py:24:60: W291 trailing whitespace
dipy/segment/quickbundles.py:25:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:30:72: W291 trailing whitespace
dipy/segment/quickbundles.py:31:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:34:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:35:80: E501 line too long (82 > 79 characters)
dipy/segment/quickbundles.py:35:83: W291 trailing whitespace
dipy/segment/quickbundles.py:36:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:38:80: E501 line too long (80 > 79 characters)
dipy/segment/quickbundles.py:45:15: E225 missing whitespace around operator
dipy/segment/quickbundles.py:45:15: E711 comparison to None should be 'if cond is not None:'
dipy/segment/quickbundles.py:45:22: W291 trailing whitespace
dipy/segment/quickbundles.py:46:25: E225 missing whitespace around operator
dipy/segment/quickbundles.py:46:43: E231 missing whitespace after ','
dipy/segment/quickbundles.py:48:25: E225 missing whitespace around operator
dipy/segment/quickbundles.py:48:32: W291 trailing whitespace
dipy/segment/quickbundles.py:49:24: E225 missing whitespace around operator
dipy/segment/quickbundles.py:50:19: E225 missing whitespace around operator
dipy/segment/quickbundles.py:51:20: E225 missing whitespace around operator
dipy/segment/quickbundles.py:51:25: W291 trailing whitespace
dipy/segment/quickbundles.py:52:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:54:22: E225 missing whitespace around operator
dipy/segment/quickbundles.py:54:22: E711 comparison to None should be 'if cond is None:'
dipy/segment/quickbundles.py:55:23: E225 missing whitespace around operator
dipy/segment/quickbundles.py:55:80: E501 line too long (112 > 79 characters)
dipy/segment/quickbundles.py:56:26: W291 trailing whitespace
dipy/segment/quickbundles.py:61:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:62:23: E231 missing whitespace after ','
dipy/segment/quickbundles.py:63:23: E225 missing whitespace around operator
dipy/segment/quickbundles.py:63:23: E711 comparison to None should be 'if cond is None:'
dipy/segment/quickbundles.py:63:30: W291 trailing whitespace
dipy/segment/quickbundles.py:64:24: E225 missing whitespace around operator
dipy/segment/quickbundles.py:65:25: E225 missing whitespace around operator
dipy/segment/quickbundles.py:66:14: E225 missing whitespace around operator
dipy/segment/quickbundles.py:67:22: E225 missing whitespace around operator
dipy/segment/quickbundles.py:67:22: E711 comparison to None should be 'if cond is None:'
dipy/segment/quickbundles.py:68:23: E225 missing whitespace around operator
dipy/segment/quickbundles.py:68:36: W291 trailing whitespace
dipy/segment/quickbundles.py:70:24: E225 missing whitespace around operator
dipy/segment/quickbundles.py:70:61: W291 trailing whitespace
dipy/segment/quickbundles.py:71:18: E225 missing whitespace around operator
dipy/segment/quickbundles.py:71:74: E231 missing whitespace after ','
dipy/segment/quickbundles.py:71:80: E501 line too long (82 > 79 characters)
dipy/segment/quickbundles.py:72:18: E225 missing whitespace around operator
dipy/segment/quickbundles.py:73:19: E225 missing whitespace around operator
dipy/segment/quickbundles.py:75:48: W291 trailing whitespace
dipy/segment/quickbundles.py:77:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:80:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:85:10: E225 missing whitespace around operator
dipy/segment/quickbundles.py:86:38: W291 trailing whitespace
dipy/segment/quickbundles.py:87:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:88:27: E231 missing whitespace after ','
dipy/segment/quickbundles.py:90:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:91:29: E231 missing whitespace after ','
dipy/segment/quickbundles.py:92:59: W291 trailing whitespace
dipy/segment/quickbundles.py:93:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:94:26: E231 missing whitespace after ','
dipy/segment/quickbundles.py:94:33: E231 missing whitespace after ','
dipy/segment/quickbundles.py:96:5: E301 expected 1 blank line, found 0
dipy/segment/quickbundles.py:96:14: W291 trailing whitespace
dipy/segment/quickbundles.py:99:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:103:35: E231 missing whitespace after ','
dipy/segment/quickbundles.py:111:10: E225 missing whitespace around operator
dipy/segment/quickbundles.py:113:25: E225 missing whitespace around operator
dipy/segment/quickbundles.py:114:25: W291 trailing whitespace
dipy/segment/quickbundles.py:115:11: E225 missing whitespace around operator
dipy/segment/quickbundles.py:116:13: E225 missing whitespace around operator
dipy/segment/quickbundles.py:118:18: E225 missing whitespace around operator
dipy/segment/quickbundles.py:119:24: E225 missing whitespace around operator
dipy/segment/quickbundles.py:120:9: E265 block comment should start with '# '
dipy/segment/quickbundles.py:121:19: E225 missing whitespace around operator
dipy/segment/quickbundles.py:122:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:123:28: E231 missing whitespace after ','
dipy/segment/quickbundles.py:126:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:127:29: E231 missing whitespace after ','
dipy/segment/quickbundles.py:130:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:134:1: W293 blank line contains whitespace
dipy/segment/quickbundles.py:138:1: W391 blank line at end of file
dipy/segment/threshold.py:3:1: E302 expected 2 blank lines, found 1
dipy/segment/benchmarks/bench_quickbundles.py:9:80: E501 line too long (80 > 79 characters)
dipy/segment/benchmarks/bench_quickbundles.py:14:80: E501 line too long (84 > 79 characters)
dipy/segment/benchmarks/bench_quickbundles.py:50:5: E265 block comment should start with '# '
dipy/segment/benchmarks/bench_quickbundles.py:72:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_clustering.py:37:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_clustering.py:38:80: E501 line too long (82 > 79 characters)
dipy/segment/tests/test_clustering.py:90:80: E501 line too long (89 > 79 characters)
dipy/segment/tests/test_clustering.py:114:80: E501 line too long (95 > 79 characters)
dipy/segment/tests/test_clustering.py:139:80: E501 line too long (82 > 79 characters)
dipy/segment/tests/test_clustering.py:144:80: E501 line too long (82 > 79 characters)
dipy/segment/tests/test_clustering.py:218:80: E501 line too long (89 > 79 characters)
dipy/segment/tests/test_clustering.py:239:80: E501 line too long (95 > 79 characters)
dipy/segment/tests/test_clustering.py:281:80: E501 line too long (97 > 79 characters)
dipy/segment/tests/test_clustering.py:286:80: E501 line too long (97 > 79 characters)
dipy/segment/tests/test_clustering.py:305:80: E501 line too long (102 > 79 characters)
dipy/segment/tests/test_clustering.py:359:80: E501 line too long (82 > 79 characters)
dipy/segment/tests/test_clustering.py:366:80: E501 line too long (97 > 79 characters)
dipy/segment/tests/test_clustering.py:391:80: E501 line too long (94 > 79 characters)
dipy/segment/tests/test_clustering.py:453:80: E501 line too long (89 > 79 characters)
dipy/segment/tests/test_clustering.py:458:80: E501 line too long (87 > 79 characters)
dipy/segment/tests/test_clustering.py:459:80: E501 line too long (84 > 79 characters)
dipy/segment/tests/test_clustering.py:460:80: E501 line too long (89 > 79 characters)
dipy/segment/tests/test_clustering.py:461:80: E501 line too long (86 > 79 characters)
dipy/segment/tests/test_clustering.py:514:5: E265 block comment should start with '# '
dipy/segment/tests/test_clustering.py:518:9: E265 block comment should start with '# '
dipy/segment/tests/test_clustering.py:549:80: E501 line too long (88 > 79 characters)
dipy/segment/tests/test_clustering.py:559:80: E501 line too long (101 > 79 characters)
dipy/segment/tests/test_clustering.py:590:80: E501 line too long (102 > 79 characters)
dipy/segment/tests/test_clustering.py:615:80: E501 line too long (95 > 79 characters)
dipy/segment/tests/test_clustering.py:619:80: E501 line too long (82 > 79 characters)
dipy/segment/tests/test_clustering.py:626:80: E501 line too long (97 > 79 characters)
dipy/segment/tests/test_clustering.py:648:80: E501 line too long (94 > 79 characters)
dipy/segment/tests/test_clustering.py:725:80: E501 line too long (91 > 79 characters)
dipy/segment/tests/test_feature.py:58:80: E501 line too long (113 > 79 characters)
dipy/segment/tests/test_feature.py:71:80: E501 line too long (91 > 79 characters)
dipy/segment/tests/test_feature.py:79:80: E501 line too long (87 > 79 characters)
dipy/segment/tests/test_feature.py:86:80: E501 line too long (91 > 79 characters)
dipy/segment/tests/test_feature.py:219:80: E501 line too long (90 > 79 characters)
dipy/segment/tests/test_feature.py:231:80: E501 line too long (127 > 79 characters)
dipy/segment/tests/test_feature.py:243:80: E501 line too long (83 > 79 characters)
dipy/segment/tests/test_feature.py:249:80: E501 line too long (127 > 79 characters)
dipy/segment/tests/test_feature.py:299:80: E501 line too long (83 > 79 characters)
dipy/segment/tests/test_mask.py:105:39: E127 continuation line over-indented for visual indent
dipy/segment/tests/test_mask.py:106:39: E127 continuation line over-indented for visual indent
dipy/segment/tests/test_mask.py:110:39: E127 continuation line over-indented for visual indent
dipy/segment/tests/test_mask.py:111:39: E127 continuation line over-indented for visual indent
dipy/segment/tests/test_mask.py:117:1: W391 blank line at end of file
dipy/segment/tests/test_metric.py:5:80: E501 line too long (83 > 79 characters)
dipy/segment/tests/test_metric.py:11:80: E501 line too long (83 > 79 characters)
dipy/segment/tests/test_metric.py:39:80: E501 line too long (81 > 79 characters)
dipy/segment/tests/test_metric.py:49:13: E731 do not assign a lambda expression, use a def
dipy/segment/tests/test_metric.py:81:80: E501 line too long (81 > 79 characters)
dipy/segment/tests/test_metric.py:82:80: E501 line too long (87 > 79 characters)
dipy/segment/tests/test_metric.py:86:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:87:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:93:80: E501 line too long (83 > 79 characters)
dipy/segment/tests/test_metric.py:107:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:162:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:163:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:185:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:229:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_metric.py:239:80: E501 line too long (81 > 79 characters)
dipy/segment/tests/test_qb.py:3:80: E501 line too long (98 > 79 characters)
dipy/segment/tests/test_qb.py:10:12: E231 missing whitespace after ','
dipy/segment/tests/test_qb.py:10:56: W291 trailing whitespace
dipy/segment/tests/test_qb.py:11:6: E225 missing whitespace around operator
dipy/segment/tests/test_qb.py:12:8: E225 missing whitespace around operator
dipy/segment/tests/test_qb.py:12:37: W291 trailing whitespace
dipy/segment/tests/test_qb.py:13:7: E225 missing whitespace around operator
dipy/segment/tests/test_qb.py:13:22: E231 missing whitespace after ','
dipy/segment/tests/test_qb.py:13:26: E231 missing whitespace after ','
dipy/segment/tests/test_qb.py:14:8: E225 missing whitespace around operator
dipy/segment/tests/test_qb.py:14:22: W291 trailing whitespace
dipy/segment/tests/test_qb.py:15:5: E265 block comment should start with '# '
dipy/segment/tests/test_qb.py:16:9: E231 missing whitespace after ','
dipy/segment/tests/test_qb.py:16:15: E225 missing whitespace around operator
dipy/segment/tests/test_qb.py:16:30: W291 trailing whitespace
dipy/segment/tests/test_qb.py:17:19: E231 missing whitespace after ','
dipy/segment/tests/test_quickbundles.py:40:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_quickbundles.py:44:80: E501 line too long (81 > 79 characters)
dipy/segment/tests/test_quickbundles.py:50:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_quickbundles.py:51:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_quickbundles.py:61:80: E501 line too long (92 > 79 characters)
dipy/segment/tests/test_quickbundles.py:75:80: E501 line too long (81 > 79 characters)
dipy/segment/tests/test_quickbundles.py:100:80: E501 line too long (80 > 79 characters)
dipy/segment/tests/test_quickbundles.py:104:80: E501 line too long (87 > 79 characters)
dipy/segment/tests/test_quickbundles.py:106:80: E501 line too long (84 > 79 characters)
dipy/segment/tests/test_quickbundles.py:118:80: E501 line too long (98 > 79 characters)
dipy/segment/tests/test_quickbundles.py:130:80: E501 line too long (96 > 79 characters)

--- a/dipy/segment/benchmarks/bench_quickbundles.py
+++ b/dipy/segment/benchmarks/bench_quickbundles.py
@@ -6,8 +6,8 @@ Run all benchmarks with::
     dipysegment.bench()
 
 If you have doctests enabled by default in nose (with a noserc file or
-environment variable), and you have a numpy version <= 1.6.1, this will also run
-the doctests, let's hope they pass.
+environment variable), and you have a numpy version <= 1.6.1, this will
+also run the doctests, let's hope they pass.
 
 Run this benchmark with:
 
@@ -47,7 +47,7 @@ def bench_quickbundles():
     fornix = [s[0].astype(dtype) for s in streams]
     fornix = streamline_utils.set_number_of_points(fornix, nb_points)
 
-    #Create eight copies of the fornix to be clustered (one in each octant).
+    # Create eight copies of the fornix to be clustered (one in each octant).
     streamlines = []
     streamlines += [s + np.array([100, 100, 100], dtype) for s in fornix]
     streamlines += [s + np.array([100, -100, 100], dtype) for s in fornix]
@@ -69,7 +69,8 @@ def bench_quickbundles():
     print("QuickBundles time: {0:.4}sec".format(qb1_time))
     assert_equal(qb.total_clusters, expected_nb_clusters)
     sizes1 = [qb.partitions()[i]['N'] for i in range(qb.total_clusters)]
-    indices1 = [qb.partitions()[i]['indices'] for i in range(qb.total_clusters)]
+    indices1 = [qb.partitions()[i]['indices']
+                for i in range(qb.total_clusters)]
 
     qb2 = QB_New(threshold)
     qb2_time = measure("clusters = qb2.cluster(streamlines)", repeat)

--- a/dipy/segment/clustering.py
+++ b/dipy/segment/clustering.py
@@ -15,6 +15,7 @@ class Identity:
     (e.g. list, tuple). Indexing an instance of this class will return the
     index provided instead of the element. It does not support slicing.
     """
+
     def __getitem__(self, idx):
         return idx
 
@@ -40,6 +41,7 @@ class Cluster(object):
     A cluster does not contain actual data but instead knows how to
     retrieve them using its `ClusterMap` object.
     """
+
     def __init__(self, id=0, indices=None, refdata=Identity()):
         self.id = id
         self.refdata = refdata
@@ -73,7 +75,8 @@ class Cluster(object):
         elif type(idx) is list:
             return [self[i] for i in idx]
 
-        raise TypeError("Index must be a int or a slice! Not " + str(type(idx)))
+        raise TypeError("Index must be a int or a slice! Not " +
+                        str(type(idx)))
 
     def __iter__(self):
         return (self[i] for i in range(len(self)))
@@ -126,6 +129,7 @@ class ClusterCentroid(Cluster):
     A cluster does not contain actual data but instead knows how to
     retrieve them using its `ClusterMapCentroid` object.
     """
+
     def __init__(self, centroid, id=0, indices=None, refdata=Identity()):
         super(ClusterCentroid, self).__init__(id, indices, refdata)
         self.centroid = centroid.copy()
@@ -175,6 +179,7 @@ class ClusterMap(object):
     refdata : list
         Actual elements that clustered indices refer to.
     """
+
     def __init__(self, refdata=Identity()):
         self._clusters = []
         self.refdata = refdata
@@ -216,7 +221,8 @@ class ClusterMap(object):
             a list of `Cluster` objects.
         """
         if isinstance(idx, np.ndarray) and idx.dtype == np.bool:
-            return [self.clusters[i] for i, take_it in enumerate(idx) if take_it]
+            return [self.clusters[i]
+                    for i, take_it in enumerate(idx) if take_it]
         elif type(idx) is slice:
             return [self.clusters[i] for i in range(*idx.indices(len(self)))]
         elif type(idx) is list:
@@ -264,12 +270,15 @@ class ClusterMap(object):
             elif op is operator.ne:
                 return not self == other
 
-            raise NotImplementedError("Can only check if two ClusterMap instances are equal or not.")
+            raise NotImplementedError(
+                "Can only check if two ClusterMap instances are equal or not.")
 
         elif isinstance(other, int):
             return np.array([op(len(cluster), other) for cluster in self])
 
-        raise NotImplementedError("ClusterMap only supports comparison with a int or another instance of Clustermap.")
+        raise NotImplementedError(
+            "ClusterMap only supports comparison with \
+            a int or another instance of Clustermap.")
 
     def __eq__(self, other):
         return self._richcmp(other, operator.eq)
@@ -401,7 +410,8 @@ class Clustering(object):
         `ClusterMap` object
             Result of the clustering.
         """
-        raise NotImplementedError("Subclass has to define method 'cluster(data, ordering)'!")
+        raise NotImplementedError(
+            "Subclass has to define method 'cluster(data, ordering)'!")
 
 
 class QuickBundles(Clustering):
@@ -422,7 +432,8 @@ class QuickBundles(Clustering):
     metric : str or `Metric` object (optional)
         The distance metric to use when comparing two streamlines. By default,
         the Minimum average Direct-Flip (MDF) distance [Garyfallidis12]_ is
-        used and streamlines are automatically resampled so they have 12 points.
+        used and streamlines are automatically resampled so they have
+        12 points.
     max_nb_clusters : int
         Limits the creation of bundles.
 
@@ -460,7 +471,9 @@ class QuickBundles(Clustering):
                         tractography simplification, Frontiers in Neuroscience,
                         vol 6, no 175, 2012.
     """
-    def __init__(self, threshold, metric="MDF_12points", max_nb_clusters=np.iinfo('i4').max):
+
+    def __init__(self, threshold, metric="MDF_12points",
+                 max_nb_clusters=np.iinfo('i4').max):
         self.threshold = threshold
         self.max_nb_clusters = max_nb_clusters
 

--- a/dipy/segment/mask.py
+++ b/dipy/segment/mask.py
@@ -234,7 +234,8 @@ def segment_from_cfa(tensor_fit, roi, threshold, return_cfa=False):
         A binary mask, which contains the bounding box for the segmentation.
 
     threshold : array-like
-        An iterable that defines the min and max values to use for the thresholding.
+        An iterable that defines the min and max values to use for the
+        thresholding.
         The values are specified as (R_min, R_max, G_min, G_max, B_min, B_max)
 
     return_cfa : bool, optional
@@ -258,7 +259,9 @@ def segment_from_cfa(tensor_fit, roi, threshold, return_cfa=False):
     cfa = color_fa(FA, tensor_fit.evecs)
     roi = np.asarray(roi, dtype=bool)
 
-    include = (cfa >= threshold[0::2]) & (cfa <= threshold[1::2]) & roi[..., None]
+    include = ((cfa >= threshold[0::2]) &
+               (cfa <= threshold[1::2]) &
+               roi[..., None])
     mask = np.all(include, axis=-1)
 
     if return_cfa:
@@ -266,9 +269,11 @@ def segment_from_cfa(tensor_fit, roi, threshold, return_cfa=False):
 
     return mask
 
+
 def clean_cc_mask(mask):
     """
-    Cleans a segmentation of the corpus callosum so no random pixels are included.
+    Cleans a segmentation of the corpus callosum so no random pixels
+    are included.
 
     Parameters
     ----------
@@ -288,7 +293,8 @@ def clean_cc_mask(mask):
     # Flood fill algorithm to find contiguous regions.
     labels, numL = label(mask)
 
-    volumes = [len(labels[np.where(labels == l_idx+1)]) for l_idx in np.arange(numL)]
+    volumes = [len(labels[np.where(labels == l_idx+1)])
+               for l_idx in np.arange(numL)]
     biggest_vol = np.arange(numL)[np.where(volumes == np.max(volumes))] + 1
     new_cc_mask[np.where(labels == biggest_vol)] = 1
 

--- a/dipy/segment/quickbundles.py
+++ b/dipy/segment/quickbundles.py
@@ -4,103 +4,109 @@ from dipy.tracking.distances import local_skeleton_clustering
 from dipy.tracking.distances import bundles_distances_mdf
 from warnings import warn
 
-warn(DeprecationWarning("Module 'dipy.segment.quickbundles' is deprecated,"
-			" use module 'dipy.segment.clustering' instead"))
+warn(DeprecationWarning("Module 'dipy.segment.quickbundles' is deprecated, \
+                         use module 'dipy.segment.clustering' instead"))
 
 
 class QuickBundles(object):
-    
-    def __init__(self,tracks,dist_thr=4.,pts=12):
-        """ Highly efficient trajectory clustering 
-        
+
+    def __init__(self, tracks, dist_thr=4., pts=12):
+        """ Highly efficient trajectory clustering
+
         Parameters
         -----------
         tracks : sequence of (N,3) ... (M,3) arrays,
                     trajectories (or tractography or streamlines)
-                    
-        dist_thr : float, 
+
+        dist_thr : float,
                     distance threshold in the space of the tracks
-        pts : int, 
-                number of points for simplifying the tracks 
-                       
+        pts : int,
+                number of points for simplifying the tracks
+
         Methods
         --------
         clustering() returns a dict holding with the clustering result
         virtuals() gives the virtuals (track centroids) of the clusters
-        exemplars() gives the exemplars (track medoids) of the clusters        
-        
+        exemplars() gives the exemplars (track medoids) of the clusters
+
         Citation
         ---------
-        
-        E.Garyfallidis, "Towards an accurate brain tractography", PhD thesis, 2012 
-        
+
+        E.Garyfallidis, "Towards an accurate brain tractography",
+        PhD thesis, 2012
+
         """
-        warn(DeprecationWarning("Class 'dipy.segment.quickbundles.QuickBundles'"
-                                " is deprecated, use module "
-                                "'dipy.segment.clustering.QuickBundles'"
-                                " instead"))
+        warn(DeprecationWarning("Class 'dipy.segment.quickbundles.QuickBundles' \
+                                is deprecated, use module \
+                                'dipy.segment.clustering.QuickBundles' \
+                                instead"))
 
         self.dist_thr = dist_thr
         self.pts = pts
-        if pts!=None:                        
-            self.tracksd=[downsample(track,self.pts) for track in tracks]
+        if pts is not None:
+            self.tracksd = [downsample(track, self.pts) for track in tracks]
         else:
-            self.tracksd=tracks                    
-        self.clustering=local_skeleton_clustering(self.tracksd, self.dist_thr)
-        self.virts=None
-        self.exemps=None                
-    
+            self.tracksd = tracks
+        self.clustering = local_skeleton_clustering(self.tracksd,
+                                                    self.dist_thr)
+        self.virts = None
+        self.exemps = None
+
     def virtuals(self):
-        if self.virts==None:
-            self.virts=[self.clustering[c]['hidden']/np.float(self.clustering[c]['N']) for c in self.clustering]
-        return self.virts      
+        if self.virts is None:
+            self.virts = [(self.clustering[c]['hidden'] /
+                           np.float(self.clustering[c]['N']))
+                          for c in self.clustering]
+        return self.virts
 
     @property
     def centroids(self):
         return self.virtuals()
-    
-    def exemplars(self,tracks=None):
-        if self.exemps==None:            
-            self.exemps=[]
-            self.exempsi=[]
-            C=self.clustering
-            if tracks==None:
-                tracks=self.tracksd            
+
+    def exemplars(self, tracks=None):
+        if self.exemps is None:
+            self.exemps = []
+            self.exempsi = []
+            C = self.clustering
+            if tracks is None:
+                tracks = self.tracksd
             for c in C:
-                cluster=[tracks[i] for i in C[c]['indices']]                
-                D=bundles_distances_mdf([C[c]['hidden']/float(C[c]['N'])],cluster)
-                D=D.ravel()
-                si=np.argmin(D)
+                cluster = [tracks[i] for i in C[c]['indices']]
+                D = bundles_distances_mdf([C[c]['hidden']/float(C[c]['N'])],
+                                          cluster)
+                D = D.ravel()
+                si = np.argmin(D)
                 self.exempsi.append(si)
-                self.exemps.append(cluster[si])                               
+                self.exemps.append(cluster[si])
         return self.exemps, self.exempsi
-    
+
     def partitions(self):
         return self.clustering
-    
+
     def clusters(self):
         return self.clustering
 
     def clusters_sizes(self):
-        C=self.clustering
-        return [C[c]['N'] for c in C] 
-    
-    def label2cluster(self,id):
+        C = self.clustering
+        return [C[c]['N'] for c in C]
+
+    def label2cluster(self, id):
         return self.clustering[id]
-    
-    def label2tracksids(self,id):
-        return [i for i in self.clustering[id]['indices']]        
-    
-    def label2tracks(self,tracks,id):
+
+    def label2tracksids(self, id):
+        return [i for i in self.clustering[id]['indices']]
+
+    def label2tracks(self, tracks, id):
         return [tracks[i] for i in self.clustering[id]['indices']]
-    @property       
+
+    @property
     def total_clusters(self):
         return len(self.clustering)
-        
+
     def downsampled_tracks(self):
         return self.tracksd
 
-    def remove_small_clusters(self,size):
+    def remove_small_clusters(self, size):
         """ Remove clusters with small size
 
         Parameters
@@ -108,31 +114,30 @@ class QuickBundles(object):
         size : int, threshold for minimum number of tracks allowed
 
         """
-        C=self.clustering
+        C = self.clustering
         for c in range(len(C)):
-            if C[c]['N']<=size:
-                del C[c]        
-        C2={}
-        keys=C.keys()
+            if C[c]['N'] <= size:
+                del C[c]
+        C2 = {}
+        keys = C.keys()
         for c in range(len(C)):
-            C2[c]=C[keys[c]]
-        self.clustering=C2
-        #self.tracksd=[downsample(track,self.pts) for track in tracks]
-        self.virts=None
-    
-    def remove_cluster(self,id):
-        print('Not implemented yet')
-        pass
-    
-    def remove_clusters(self,list_ids):
-        print('Not implemented yet')
-        pass
-    
-    def remove_tracks(self):
-        print('Not implemented yet')
-        pass
-    
-    def points_per_track(self):
+            C2[c] = C[keys[c]]
+        self.clustering = C2
+        # self.tracksd=[downsample(track,self.pts) for track in tracks]
+        self.virts = None
+
+    def remove_cluster(self, id):
         print('Not implemented yet')
         pass
 
+    def remove_clusters(self, list_ids):
+        print('Not implemented yet')
+        pass
+
+    def remove_tracks(self):
+        print('Not implemented yet')
+        pass
+
+    def points_per_track(self):
+        print('Not implemented yet')
+        pass

--- a/dipy/segment/tests/test_clustering.py
+++ b/dipy/segment/tests/test_clustering.py
@@ -34,8 +34,12 @@ def test_cluster_attributes_and_constructor():
     assert_equal(len(cluster), 0)
 
     # Duplicate
-    assert_equal(cluster, Cluster(cluster.id, cluster.indices, cluster.refdata))
-    assert_false(cluster != Cluster(cluster.id, cluster.indices, cluster.refdata))
+    assert_equal(cluster, Cluster(cluster.id,
+                                  cluster.indices,
+                                  cluster.refdata))
+    assert_false(cluster != Cluster(cluster.id,
+                                    cluster.indices,
+                                    cluster.refdata))
 
     # Invalid comparison
     assert_raises(TypeError, cluster.__cmp__, cluster)
@@ -87,7 +91,8 @@ def test_cluster_getitem():
         assert_equal(cluster[i], indices[i])
 
     # Test advanced indexing
-    assert_array_equal(cluster[advanced_indices], [indices[i] for i in advanced_indices])
+    assert_array_equal(cluster[advanced_indices],
+                       [indices[i] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster.__getitem__, len(cluster))
@@ -111,7 +116,8 @@ def test_cluster_getitem():
         assert_array_equal(cluster[i], data[indices[i]])
 
     # Test advanced indexing
-    assert_array_equal(cluster[advanced_indices], [data[indices[i]] for i in advanced_indices])
+    assert_array_equal(cluster[advanced_indices],
+                       [data[indices[i]] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster.__getitem__, len(cluster))
@@ -136,12 +142,14 @@ def test_cluster_str_and_repr():
     cluster = Cluster()
     cluster.assign(*indices)
     assert_equal(str(cluster), "[" + ", ".join(map(str, indices)) + "]")
-    assert_equal(repr(cluster), "Cluster([" + ", ".join(map(str, indices)) + "])")
+    assert_equal(repr(cluster),
+                 "Cluster([" + ", ".join(map(str, indices)) + "])")
 
     # Test with specifying refdata in ClusterMap
     cluster.refdata = data
     assert_equal(str(cluster), "[" + ", ".join(map(str, indices)) + "]")
-    assert_equal(repr(cluster), "Cluster([" + ", ".join(map(str, indices)) + "])")
+    assert_equal(repr(cluster),
+                 "Cluster([" + ", ".join(map(str, indices)) + "])")
 
 
 def test_cluster_centroid_attributes_and_constructor():
@@ -215,7 +223,8 @@ def test_cluster_centroid_getitem():
         assert_equal(cluster[i], indices[i])
 
     # Test advanced indexing
-    assert_array_equal(cluster[advanced_indices], [indices[i] for i in advanced_indices])
+    assert_array_equal(cluster[advanced_indices],
+                       [indices[i] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster.__getitem__, len(cluster))
@@ -236,7 +245,8 @@ def test_cluster_centroid_getitem():
         assert_array_equal(cluster[i], data[indices[i]])
 
     # Test advanced indexing
-    assert_array_equal(cluster[advanced_indices], [data[indices[i]] for i in advanced_indices])
+    assert_array_equal(cluster[advanced_indices],
+                       [data[indices[i]] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster.__getitem__, len(cluster))
@@ -278,12 +288,14 @@ def test_cluster_map_add_cluster():
         assert_equal(len(clusters), i+1)
         assert_equal(cluster, clusters[-1])
 
-    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*list_of_indices)))
+    assert_array_equal(list(itertools.chain(*clusters)),
+                       list(itertools.chain(*list_of_indices)))
 
     # Test adding multiple clusters at once.
     clusters = ClusterMap()
     clusters.add_cluster(*list_of_cluster_objects)
-    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*list_of_indices)))
+    assert_array_equal(list(itertools.chain(*clusters)),
+                       list(itertools.chain(*list_of_indices)))
 
 
 def test_cluster_map_remove_cluster():
@@ -302,7 +314,8 @@ def test_cluster_map_remove_cluster():
 
     clusters.remove_cluster(cluster2)
     assert_equal(len(clusters), 2)
-    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*[cluster1, cluster3])))
+    assert_array_equal(list(itertools.chain(*clusters)),
+                       list(itertools.chain(*[cluster1, cluster3])))
     assert_equal(clusters[0], cluster1)
     assert_equal(clusters[1], cluster3)
 
@@ -356,14 +369,16 @@ def test_cluster_map_iter():
         cluster_map.add_cluster(new_cluster)
         clusters.append(new_cluster)
 
-    assert_true(all([c1 is c2 for c1, c2 in zip(cluster_map.clusters, clusters)]))
+    assert_true(all([c1 is c2 for c1, c2 in zip(cluster_map.clusters,
+                                                clusters)]))
     assert_array_equal(cluster_map, clusters)
     assert_array_equal(cluster_map.clusters, clusters)
     assert_array_equal(cluster_map, [cluster.indices for cluster in clusters])
 
     # Set refdata
     cluster_map.refdata = data
-    assert_array_equal(cluster_map, [[data[i] for i in cluster.indices] for cluster in clusters])
+    assert_array_equal(cluster_map, [[data[i] for i in cluster.indices]
+                                     for cluster in clusters])
 
     # Remove refdata, i.e. back to indices
     cluster_map.refdata = None
@@ -388,7 +403,8 @@ def test_cluster_map_getitem():
         assert_equal(cluster_map[i], clusters[i])
 
     # Test advanced indexing
-    assert_array_equal(cluster_map[advanced_indices], [clusters[i] for i in advanced_indices])
+    assert_array_equal(cluster_map[advanced_indices],
+                       [clusters[i] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster_map.__getitem__, len(clusters))
@@ -450,15 +466,20 @@ def test_cluster_map_get_small_and_large_clusters():
     cluster_map.add_cluster(*small_clusters)
 
     # Randomly generate small clusters
-    indices = [rng.randint(0, 10, size=i) for i in range(nb_clusters+1, 2*nb_clusters+1)]
+    indices = [rng.randint(0, 10, size=i)
+               for i in range(nb_clusters+1, 2*nb_clusters+1)]
     large_clusters = [Cluster(indices=indices[i]) for i in range(nb_clusters)]
     cluster_map.add_cluster(*large_clusters)
 
     assert_equal(len(cluster_map), 2*nb_clusters)
-    assert_equal(len(cluster_map.get_small_clusters(nb_clusters)), len(small_clusters))
-    assert_arrays_equal(cluster_map.get_small_clusters(nb_clusters), small_clusters)
-    assert_equal(len(cluster_map.get_large_clusters(nb_clusters+1)), len(large_clusters))
-    assert_arrays_equal(cluster_map.get_large_clusters(nb_clusters+1), large_clusters)
+    assert_equal(len(cluster_map.get_small_clusters(nb_clusters)),
+                 len(small_clusters))
+    assert_arrays_equal(cluster_map.get_small_clusters(nb_clusters),
+                        small_clusters)
+    assert_equal(len(cluster_map.get_large_clusters(nb_clusters+1)),
+                 len(large_clusters))
+    assert_arrays_equal(cluster_map.get_large_clusters(nb_clusters+1),
+                        large_clusters)
 
 
 def test_cluster_map_comparison_with_int():
@@ -511,11 +532,11 @@ def test_cluster_map_comparison_with_int():
 def test_cluster_map_comparison_with_object():
     nb_clusters = 4
     cluster_map = ClusterMap()
-    #clusters = []
+    # clusters = []
     for i in range(nb_clusters):
         new_cluster = Cluster(indices=range(i))
         cluster_map.add_cluster(new_cluster)
-        #clusters.append(new_cluster)
+        # clusters.append(new_cluster)
 
     # Comparison with another ClusterMap object
     other_cluster_map = copy.deepcopy(cluster_map)
@@ -546,7 +567,8 @@ def test_cluster_map_centroid_add_cluster():
 
         centroids.append(np.zeros_like(features))
         for id_data in range(2*i):
-            centroids[-1] = (centroids[-1]*id_data + (id_data+1)*features) / (id_data+1)
+            centroids[-1] = ((centroids[-1]*id_data + (id_data+1)*features) /
+                             (id_data+1))
             cluster.assign(id_data, (id_data+1)*features)
             cluster.update()
 
@@ -556,7 +578,8 @@ def test_cluster_map_centroid_add_cluster():
         assert_equal(cluster, clusters[-1])
 
     assert_equal(type(clusters.centroids), list)
-    assert_array_equal(list(itertools.chain(*clusters.centroids)), list(itertools.chain(*centroids)))
+    assert_array_equal(list(itertools.chain(*clusters.centroids)),
+                       list(itertools.chain(*centroids)))
 
     # Check adding features of different sizes (shorter and longer)
     features_shape_short = (1, features_shape[1]-3)
@@ -587,7 +610,8 @@ def test_cluster_map_centroid_remove_cluster():
 
     clusters.remove_cluster(cluster2)
     assert_equal(len(clusters), 2)
-    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*[cluster1, cluster3])))
+    assert_array_equal(list(itertools.chain(*clusters)),
+                       list(itertools.chain(*[cluster1, cluster3])))
     assert_array_equal(clusters.centroids, np.array([centroid1, centroid3]))
     assert_equal(clusters[0], cluster1)
     assert_equal(clusters[1], cluster3)
@@ -612,18 +636,22 @@ def test_cluster_map_centroid_iter():
     clusters = []
     for i in range(nb_clusters):
         new_centroid = np.zeros_like(features)
-        new_cluster = ClusterCentroid(new_centroid, indices=rng.randint(0, len(data), size=10))
+        new_cluster = ClusterCentroid(new_centroid,
+                                      indices=rng.randint(0, len(data),
+                                                          size=10))
         cluster_map.add_cluster(new_cluster)
         clusters.append(new_cluster)
 
-    assert_true(all([c1 is c2 for c1, c2 in zip(cluster_map.clusters, clusters)]))
+    assert_true(all([c1 is c2 for c1, c2 in
+                     zip(cluster_map.clusters, clusters)]))
     assert_array_equal(cluster_map, clusters)
     assert_array_equal(cluster_map.clusters, clusters)
     assert_array_equal(cluster_map, [cluster.indices for cluster in clusters])
 
     # Set refdata
     cluster_map.refdata = data
-    assert_array_equal(cluster_map, [[data[i] for i in cluster.indices] for cluster in clusters])
+    assert_array_equal(cluster_map, [[data[i] for i in cluster.indices]
+                                     for cluster in clusters])
 
 
 def test_cluster_map_centroid_getitem():
@@ -645,7 +673,8 @@ def test_cluster_map_centroid_getitem():
         assert_equal(cluster_map[i], clusters[i])
 
     # Test advanced indexing
-    assert_array_equal(cluster_map[advanced_indices], [clusters[i] for i in advanced_indices])
+    assert_array_equal(cluster_map[advanced_indices],
+                       [clusters[i] for i in advanced_indices])
 
     # Test index out of bounds
     assert_raises(IndexError, cluster_map.__getitem__, len(clusters))
@@ -722,7 +751,9 @@ def test_subclassing_clustering():
             pass
 
     clustering_algo = SubClustering()
-    assert_raises(NotImplementedError, super(SubClustering, clustering_algo).cluster, None)
+    assert_raises(NotImplementedError,
+                  super(SubClustering, clustering_algo).cluster,
+                  None)
 
 
 if __name__ == '__main__':

--- a/dipy/segment/tests/test_feature.py
+++ b/dipy/segment/tests/test_feature.py
@@ -55,7 +55,8 @@ def test_feature_resample():
             super(ResampleFeature, self).__init__(is_order_invariant=False)
             self.nb_points = nb_points
             if nb_points <= 0:
-                raise ValueError("ResampleFeature: `nb_points` must be strictly positive: {0}".format(nb_points))
+                raise ValueError("ResampleFeature: `nb_points` must be \
+                                  strictly positive: {0}".format(nb_points))
 
         def infer_shape(self, streamline):
             return (self.nb_points, streamline.shape[1])
@@ -68,7 +69,8 @@ def test_feature_resample():
 
     max_points = max(map(len, [s1, s2, s3, s4]))
     for nb_points in [2, 5, 2*max_points]:
-        for feature in [dipymetric.ResampleFeature(nb_points), ResampleFeature(nb_points)]:
+        for feature in [dipymetric.ResampleFeature(nb_points),
+                        ResampleFeature(nb_points)]:
             for s in [s1, s2, s3, s4]:
                 # Test method infer_shape
                 assert_equal(feature.infer_shape(s), (nb_points, s.shape[1]))
@@ -76,14 +78,16 @@ def test_feature_resample():
                 # Test method extract
                 features = feature.extract(s)
                 assert_equal(features.shape, (nb_points, s.shape[1]))
-                assert_array_almost_equal(features, set_number_of_points(s, nb_points))
+                assert_array_almost_equal(features,
+                                          set_number_of_points(s, nb_points))
 
             # This feature type is not order invariant
             assert_false(feature.is_order_invariant)
             for s in [s1, s2, s3, s4]:
                 features = feature.extract(s)
                 features_flip = feature.extract(s[::-1])
-                assert_array_equal(features_flip, set_number_of_points(s[::-1], nb_points))
+                assert_array_equal(features_flip,
+                                   set_number_of_points(s[::-1], nb_points))
                 assert_true(np.any(np.not_equal(features, features_flip)))
 
 
@@ -216,7 +220,8 @@ def test_feature_vector_of_endpoints():
 
 
 def test_feature_extract():
-    # Test that features are automatically cast into float32 when coming from Python space
+    # Test that features are automatically cast into float32 when
+    # coming from Python space
     class CenterOfMass64bit(dipymetric.Feature):
         def infer_shape(self, streamline):
             return streamline.shape[1]
@@ -228,7 +233,9 @@ def test_feature_extract():
     feature_shape = (1, 3)  # One N-dimensional point
     feature = CenterOfMass64bit()
 
-    streamlines = [np.arange(np.random.randint(20, 30) * 3).reshape((-1, 3)).astype(np.float32) for i in range(nb_streamlines)]
+    streamlines = [np.arange(np.random.randint(20, 30) * 3)
+                   .reshape((-1, 3)).astype(np.float32)
+                   for i in range(nb_streamlines)]
     features = extract(feature, streamlines)
 
     assert_equal(len(features), len(streamlines))
@@ -240,13 +247,16 @@ def test_feature_extract():
             return 1
 
         def extract(self, streamline):
-            return np.sum(np.sqrt(np.sum((streamline[1:] - streamline[:-1]) ** 2)))
+            return np.sum(np.sqrt(np.sum((streamline[1:] -
+                                          streamline[:-1]) ** 2)))
 
     nb_streamlines = 100
     feature_shape = (1, 1)  # One scalar represented as a 2D array
     feature = ArcLengthFeature()
 
-    streamlines = [np.arange(np.random.randint(20, 30) * 3).reshape((-1, 3)).astype(np.float32) for i in range(nb_streamlines)]
+    streamlines = [np.arange(np.random.randint(20, 30) * 3)
+                   .reshape((-1, 3)).astype(np.float32)
+                   for i in range(nb_streamlines)]
     features = extract(feature, streamlines)
 
     assert_equal(len(features), len(streamlines))
@@ -296,7 +306,8 @@ def test_using_python_feature_with_cython_metric():
             return long(1)
 
         def extract(self, streamline):
-            return np.sum(np.sqrt(np.sum((streamline[1:] - streamline[:-1]) ** 2)))
+            return np.sum(np.sqrt(np.sum((streamline[1:] -
+                                          streamline[:-1]) ** 2)))
 
     # Test using Python Feature with Cython Metric
     feature = ArcLengthFeature()

--- a/dipy/segment/tests/test_mask.py
+++ b/dipy/segment/tests/test_mask.py
@@ -102,16 +102,15 @@ def test_median_otsu():
     assert_almost_equal(mask.sum(), mask2.sum())
 
     _, mask3 = median_otsu(data2, median_radius=3, numpass=2,
-                                      autocrop=False, vol_idx=[0, 1],
-                                      dilate=1)
+                           autocrop=False, vol_idx=[0, 1],
+                           dilate=1)
     assert_equal(mask2.sum() < mask3.sum(), True)
 
     _, mask4 = median_otsu(data2, median_radius=3, numpass=2,
-                                      autocrop=False, vol_idx=[0, 1],
-                                      dilate=2)
+                           autocrop=False, vol_idx=[0, 1],
+                           dilate=2)
     assert_equal(mask3.sum() < mask4.sum(), True)
 
 
 if __name__ == '__main__':
     run_module_suite()
-

--- a/dipy/segment/tests/test_metric.py
+++ b/dipy/segment/tests/test_metric.py
@@ -2,13 +2,15 @@ import numpy as np
 import dipy.segment.metric as dipymetric
 import itertools
 
-from nose.tools import assert_true, assert_false, assert_equal, assert_almost_equal
+from nose.tools import (assert_true, assert_false,
+                        assert_equal, assert_almost_equal)
 from numpy.testing import assert_array_equal, assert_raises, run_module_suite
 
 
 def norm(x, ord=None, axis=None):
     if axis is not None:
-        return np.apply_along_axis(np.linalg.norm, axis, x.astype(np.float64), ord)
+        return np.apply_along_axis(np.linalg.norm, axis,
+                                   x.astype(np.float64), ord)
 
     return np.linalg.norm(x.astype(np.float64), ord=ord)
 
@@ -36,7 +38,8 @@ def test_metric_minimum_average_direct_flip():
 
     class MinimumAverageDirectFlipMetric(dipymetric.Metric):
         def __init__(self, feature):
-            super(MinimumAverageDirectFlipMetric, self).__init__(feature=feature)
+            super(MinimumAverageDirectFlipMetric, self).__init__(
+                    feature=feature)
 
         @property
         def is_order_invariant(self):
@@ -46,7 +49,8 @@ def test_metric_minimum_average_direct_flip():
             return shape1[0] == shape2[0]
 
         def dist(self, v1, v2):
-            average_euclidean = lambda x, y: np.mean(norm(x-y, axis=1))
+            def average_euclidean(x, y):
+                return np.mean(norm(x-y, axis=1))
             dist_direct = average_euclidean(v1, v2)
             dist_flipped = average_euclidean(v1, v2[::-1])
             return min(dist_direct, dist_flipped)
@@ -78,19 +82,24 @@ def test_metric_minimum_average_direct_flip():
         s_zero_mean = s - s_mean
         s_rotated = np.dot(M_rotation, s_zero_mean.T).T + s_mean
 
-        opposite = norm(np.cross(rot_axis, s_zero_mean), axis=1) / norm(rot_axis)
-        distances = np.sqrt(2*opposite**2 * (1 - np.cos(60.*np.pi/180.))).astype(dtype)
+        opposite = norm(np.cross(rot_axis, s_zero_mean),
+                        axis=1) / norm(rot_axis)
+        distances = np.sqrt(2*opposite**2 *
+                            (1 - np.cos(60.*np.pi/180.))).astype(dtype)
         d = np.mean(distances)
         assert_almost_equal(metric.dist(s, s_rotated), d, 5)
 
-        for s1, s2 in itertools.product(*[streamlines]*2):  # All possible pairs
-            # Extract features since metric doesn't work directly on streamlines
+        # All possible pairs
+        for s1, s2 in itertools.product(*[streamlines]*2):
+            # Extract features since metric doesn't work
+            # directly on streamlines
             f1 = metric.feature.extract(s1)
             f2 = metric.feature.extract(s2)
 
             # Test method are_compatible
             same_nb_points = f1.shape[0] == f2.shape[0]
-            assert_equal(metric.are_compatible(f1.shape, f2.shape), same_nb_points)
+            assert_equal(metric.are_compatible(f1.shape, f2.shape),
+                         same_nb_points)
 
             # Test method dist if features are compatible
             if metric.are_compatible(f1.shape, f2.shape):
@@ -104,7 +113,8 @@ def test_metric_minimum_average_direct_flip():
 
         # This metric type is order invariant
         assert_true(metric.is_order_invariant)
-        for s1, s2 in itertools.product(*[streamlines]*2):  # All possible pairs
+        # All possible pairs
+        for s1, s2 in itertools.product(*[streamlines]*2):
             f1 = metric.feature.extract(s1)
             f2 = metric.feature.extract(s2)
 
@@ -159,8 +169,10 @@ def test_metric_cosine():
         assert_equal(metric.dist(v1, v2), 0.5)  # orthogonal
         assert_equal(metric.dist(v1, v3), 1.)   # opposite
 
-        for s1, s2 in itertools.product(*[streamlines]*2):  # All possible pairs
-            # Extract features since metric doesn't work directly on streamlines
+        # All possible pairs
+        for s1, s2 in itertools.product(*[streamlines]*2):
+            # Extract features since metric doesn't
+            # work directly on streamlines
             f1 = metric.feature.extract(s1)
             f2 = metric.feature.extract(s2)
 
@@ -182,7 +194,8 @@ def test_metric_cosine():
 
         # This metric type is not order invariant
         assert_false(metric.is_order_invariant)
-        for s1, s2 in itertools.product(*[streamlines]*2):  # All possible pairs
+        # All possible pairs
+        for s1, s2 in itertools.product(*[streamlines]*2):
             f1 = metric.feature.extract(s1)
             f2 = metric.feature.extract(s2)
 
@@ -226,7 +239,8 @@ def test_distance_matrix():
 
         for i in range(len(data)):
             for j in range(len(data)):
-                assert_equal(D[i, j], dipymetric.dist(metric, data[i], data[j]))
+                assert_equal(D[i, j], dipymetric.dist(metric, data[i],
+                                                      data[j]))
 
         # Compute distances of all tuples spawn by the Cartesian product
         # of `data` with `data2`.
@@ -236,7 +250,8 @@ def test_distance_matrix():
 
         for i in range(len(data)):
             for j in range(len(data2)):
-                assert_equal(D[i, j], dipymetric.dist(metric, data[i], data2[j]))
+                assert_equal(D[i, j], dipymetric.dist(metric, data[i],
+                                                      data2[j]))
 
 
 if __name__ == '__main__':

--- a/dipy/segment/tests/test_qb.py
+++ b/dipy/segment/tests/test_qb.py
@@ -1,17 +1,18 @@
 import numpy as np
 import nibabel as nib
-from nose.tools import assert_true, assert_false, assert_equal, assert_almost_equal, assert_raises
+from nose.tools import (assert_true, assert_false, assert_equal,
+                        assert_almost_equal, assert_raises)
 from numpy.testing import assert_array_equal, assert_array_almost_equal
 from dipy.data import get_data
 from dipy.segment.quickbundles import QuickBundles
 
 
 def test_qbundles():
-    streams,hdr = nib.trackvis.read(get_data('fornix'))    
-    T=[s[0] for s in streams]
-    Trk=np.array(T, dtype=np.object)    
-    qb=QuickBundles(T,10.,12)
-    Tqb=qb.virtuals()    
-    #Tqbe,Tqbei=qb.exemplars(T)
-    Tqbe,Tqbei=qb.exemplars()    
-    assert_equal(4,qb.total_clusters)
+    streams, hdr = nib.trackvis.read(get_data('fornix'))
+    T = [s[0] for s in streams]
+    Trk = np.array(T, dtype=np.object)
+    qb = QuickBundles(T, 10., 12)
+    Tqb = qb.virtuals()
+    # Tqbe,Tqbei=qb.exemplars(T)
+    Tqbe, Tqbei = qb.exemplars()
+    assert_equal(4, qb.total_clusters)

--- a/dipy/segment/tests/test_quickbundles.py
+++ b/dipy/segment/tests/test_quickbundles.py
@@ -37,19 +37,20 @@ def test_quickbundles_empty_data():
 
 
 def test_quickbundles_wrong_metric():
-    assert_raises(ValueError, QuickBundles, threshold=10., metric="WrongMetric")
+    assert_raises(ValueError, QuickBundles,
+                  threshold=10., metric="WrongMetric")
 
 
 def test_quickbundles_shape_uncompatibility():
-    # QuickBundles' old default metric (AveragePointwiseEuclideanMetric, aka MDF)
-    # requires that all streamlines have the same number of points.
+    # QuickBundles' old default metric (AveragePointwiseEuclideanMetric,
+    #  aka MDF) requires that all streamlines have the same number of points.
     metric = dipymetric.AveragePointwiseEuclideanMetric()
     qb = QuickBundles(threshold=20., metric=metric)
     assert_raises(ValueError, qb.cluster, data)
 
-    # QuickBundles' new default metric (AveragePointwiseEuclideanMetric, aka MDF
-    #  combined with ResampleFeature) will automatically resample streamlines so
-    #  they all have 18 points.
+    # QuickBundles' new default metric (AveragePointwiseEuclideanMetric,
+    # aka MDF combined with ResampleFeature) will automatically resample
+    # streamlines so they all have 18 points.
     qb = QuickBundles(threshold=20.)
     clusters1 = qb.cluster(data)
 
@@ -58,7 +59,8 @@ def test_quickbundles_shape_uncompatibility():
     qb = QuickBundles(threshold=20., metric=metric)
     clusters2 = qb.cluster(data)
 
-    assert_array_equal(list(itertools.chain(*clusters1)), list(itertools.chain(*clusters2)))
+    assert_array_equal(list(itertools.chain(*clusters1)),
+                       list(itertools.chain(*clusters2)))
 
 
 def test_quickbundles_2D():
@@ -72,7 +74,8 @@ def test_quickbundles_2D():
     data += [rng.randn(1, 2) + np.array([-10, -10]) for i in range(5)]
     data = np.array(data, dtype=dtype)
 
-    clusters_truth = [[0], [1, 2], [3, 4, 5], [6, 7, 8, 9], [10, 11, 12, 13, 14]]
+    clusters_truth = [[0], [1, 2], [3, 4, 5],
+                      [6, 7, 8, 9], [10, 11, 12, 13, 14]]
 
     # # Uncomment the following to visualize this test
     # import pylab as plt
@@ -97,13 +100,16 @@ def test_quickbundles_2D():
             # Find the corresponding cluster in 'clusters_truth'
             for cluster_truth in clusters_truth:
                 if cluster_truth[0] in cluster.indices:
-                    assert_equal(sorted(cluster.indices), sorted(cluster_truth))
+                    assert_equal(sorted(cluster.indices),
+                                 sorted(cluster_truth))
 
     # Cluster each cluster again using a small threshold
     for cluster in clusters:
-        subclusters = quickbundles(data, metric, threshold=0, ordering=cluster.indices)
+        subclusters = quickbundles(data, metric, threshold=0,
+                                   ordering=cluster.indices)
         assert_equal(len(subclusters), len(cluster))
-        assert_equal(sorted(itertools.chain(*subclusters)), sorted(cluster.indices))
+        assert_equal(sorted(itertools.chain(*subclusters)),
+                     sorted(cluster.indices))
 
     # A very large threshold should produce only 1 cluster
     clusters = quickbundles(data, metric, threshold=np.inf)
@@ -115,7 +121,8 @@ def test_quickbundles_2D():
     clusters = quickbundles(data, metric, threshold=0)
     assert_equal(len(clusters), len(data))
     assert_array_equal(list(map(len, clusters)), np.ones(len(data)))
-    assert_array_equal([idx for cluster in clusters for idx in cluster.indices], range(len(data)))
+    assert_array_equal([idx for cluster in clusters
+                        for idx in cluster.indices], range(len(data)))
 
 
 def test_quickbundles_streamlines():
@@ -127,7 +134,8 @@ def test_quickbundles_streamlines():
     assert_equal(clusters.refdata, rdata)
     # Set `refdata` to return indices instead of actual data points.
     clusters.refdata = None
-    assert_array_equal(list(itertools.chain(*clusters)), list(itertools.chain(*clusters_truth)))
+    assert_array_equal(list(itertools.chain(*clusters)),
+                       list(itertools.chain(*clusters_truth)))
 
     # Cluster read-only data
     for datum in rdata:

--- a/dipy/segment/threshold.py
+++ b/dipy/segment/threshold.py
@@ -1,5 +1,6 @@
 import numpy as np
 
+
 def otsu(image, nbins=256):
     """
     Return threshold value based on Otsu's method.

